app/gui/view/graph-editor/src/lib.rs

```diff
@@ -3901,3 +3917,98 @@ mod tests {
         (app, graph_editor)
     }
 }
+
+
+
+// ==================
+// === Automation ===
+// ==================
+
+/// Utilities for simulating user interaction, to support testing and profiling.
+// This is not #[cfg(test)] because it is needed for batch-mode profiling,
+// which is not just in testing builds.
+pub mod automation {
+    use super::*;
+    use enso_frp::future::EventOutputExt;
+
+
+
+    // =====================
+    // === Node creation ===
+    // =====================
+
+    // FIXME: This functionality overlaps with the [`tests::add_node_by`] functions above.
```

Comment:
The duplication was already present, I just added the note. I can't refactor it right now because I can't get the integration tests to work. I opened a bug (https://www.pivotaltracker.com/story/show/182278293) and added a link in the comment.